nbs/00_vector.ipynb

```diff
@@ -296,12 +296,46 @@
    "source": [
     "#| export\n",
     "class UUIDTimeRange:\n",
+    "    \n",
+    "    @staticmethod\n",
+    "    def _parse_datetime(input_datetime: Union[datetime, str]):\n",
+    "        \"\"\"\n",
+    "        Parse a datetime object or string representation of a datetime.\n",
+    "\n",
+    "        Args:\n",
+    "            input_datetime (datetime or str): Input datetime or string.\n",
+    "\n",
+    "        Returns:\n",
+    "            datetime: Parsed datetime object.\n",
+    "\n",
+    "        Raises:\n",
+    "            ValueError: If the input cannot be parsed as a datetime.\n",
+    "        \"\"\"\n",
+    "        if input_datetime is None or input_datetime == \"None\":\n",
+    "            return None\n",
+    "        \n",
+    "        if isinstance(input_datetime, datetime):\n",
+    "            # If input is already a datetime object, return it as is\n",
+    "            return input_datetime\n",
+    "\n",
+    "        if isinstance(input_datetime, str):\n",
+    "            try:\n",
+    "                # Attempt to parse the input string into a datetime\n",
+    "                return datetime.fromisoformat(input_datetime)\n",
+    "            except ValueError:\n",
+    "                raise ValueError(\"Invalid datetime string format: {}\".format(input_datetime))\n",
+    "\n",
+    "        raise ValueError(\"Input must be a datetime object or string\")\n",
+    "\n",
     "    def __init__(self, start_date: Optional[datetime] = None, end_date: Optional[datetime] = None, time_delta: Optional[timedelta] = None, start_inclusive=True, end_inclusive=False):\n",
```

Comment:
should the types now be `Optional[Union[datetime, str]]` for `start_date` and `end_date` now?

Reply:
Good catch